components/ws-daemon/pkg/iws/iws.go

```diff
@@ -195,16 +198,54 @@ func (wbs *InWorkspaceServiceServer) PrepareForUserNS(ctx context.Context, req *
 		return nil, status.Errorf(codes.Internal, "cannot find workspace rootfs")
 	}
 
+	log.WithField("type", wbs.FSShift).Debug("FSShift")
+
+	// user namespace support for FUSE landed in Linux 4.18:
+	//   - http://lkml.iu.edu/hypermail/linux/kernel/1806.0/04385.html
+	// Development leading up to this point:
+	//   - https://lists.linuxcontainers.org/pipermail/lxc-devel/2014-July/009797.html
+	//   - https://lists.linuxcontainers.org/pipermail/lxc-users/2014-October/007948.html
 	err = nsinsider(wbs.Session.InstanceID, int(containerPID), func(c *exec.Cmd) {
+		c.Args = append(c.Args, "mknod-fuse")
+	})
+	if err != nil {
+		log.WithError(err).WithFields(wbs.Session.OWI()).Error("PrepareForUserNS: cannot mknod fuse")
+		return nil, status.Errorf(codes.Internal, "cannot prepare FUSE")
+	}
+
+	_ = os.MkdirAll(filepath.Join(wbs.Session.ServiceLocDaemon, "mark"), 0755)
+	mountpoint := filepath.Join(wbs.Session.ServiceLocNode, "mark")
+
+	if wbs.FSShift == api.FSShiftMethod_FUSE {
+		err = nsinsider(wbs.Session.InstanceID, int(1), func(c *exec.Cmd) {
+			mappings := fmt.Sprintf("0:%v:1:1:100000:65534", wsinit.GitpodUID)
```

Comment:
nit: I'm wondering if there's a good way to make the ID mapping more explicit here. E.g. make it a constant, or at least a rather explicit comment :)